community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -496,19 +497,37 @@ static DurationValue durationBetween( Temporal from, Temporal to )
         long days = 0;
         if ( from.isSupported( EPOCH_DAY ) && to.isSupported( EPOCH_DAY ) )
         {
-            Period period = Period.between( LocalDate.from( from ), LocalDate.from( to ) );
+            LocalDate fromDate;
+            LocalDate toDate;
+            try
+            {
+                fromDate = LocalDate.from( from );
+                toDate = LocalDate.from( to );
+            }
+            catch ( DateTimeException e )
+            {
+                throw new InvalidTemporalArgumentException( e.getMessage(), e );
+            }
+            Period period = Period.between( fromDate, toDate );
             months = period.getYears() * 12L + period.getMonths();
             days = period.getDays();
             if ( months != 0 || days != 0 )
             {
                 // Adjust in order to get to a point where we can compute the time difference,
                 // without having to bother with the length of days (which might differ due to timezone)
-                from = from.plus( period );
+                try
+                {
+                    from = from.plus( period );
```

Comment:
Is this try catch not equivalent to using `assertValidArithmetic`?

Reply:
It is, but `assertValidArithmetic` is a method of `TemporalValue` and a `DurationValue` is not a `TemporalValue`. I could of course copy the method to this file, but it would only be used once, so I decided to inline it instead.